cairo/tests/programs/test_os.py

```diff
@@ -121,3 +121,40 @@ def test_block_hint(self, cairo_run):
                 else []
             ),
         ]
+
+    def test_create_tx_returndata(self, cairo_run):
+        initial_state = {
+            OWNER: {
+                "code": [],
+                "storage": {},
+                "balance": int(1e18),
+                "nonce": 0,
+            },
+            COINBASE: {
+                "code": [],
+                "storage": {},
+                "balance": 0,
+                "nonce": 0,
+            },
+            "0x32dCAB0EF3FB2De2fce1D2E0799D36239671F04A": {
+                "code": [],
+                "storage": {},
+                "balance": 0,
+                "nonce": 0,
+            },
+        }
+        transaction = {
+            "to": "",
+            "data": "0x604580600e600039806000f350fe7fffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffe03601600081602082378035828234f58015156039578182fd5b8082525050506014600cf3",
+            "value": 0,
+            "signer": OWNER,
+        }
+        state = cairo_run(
```

Comment:
This test only ensures the state is correct but do not verify the return data.
As the execution data is not needed in the context of keth, it seems difficult to test. Modifying the os to get it does not make sense either: adds code and complexity that will not be used